ui/src/api/login.js

```diff
@@ -0,0 +1,23 @@
+import api from '.';
+
+const endpoint = api.url('/login');
```

Comment:
Here, I prepare another `wretch` instance (based on the base instance) configured specifically for the `/login` endpoint. The rest of this file is hopefully straightforward, but it's good to know a bit how wretch works:

- `get()`, `post()`, etc. call `fetch()` internally and return a `WretchResponseChain` object.
- To get the response data, we can chain this object with `json()` if the endpoint responds with JSON, or `blob()`, `text()`, etc.
- After that, we get a normal `Promise` with the data, which we can `await` or chain with `then()`/`catch()` as usual.

For endpoints that don't respond with any data, we call `.res()` (short for "response") instead of `json()` to ensure we still return a `Promise` (but with `res()`, the promise resolves to the raw `Response` object).

Reply:
Ok, nice :)